File: src/main/java/com/practica/crean/arquitecture/core/usecases/impl/BuscarEstudiantesImpl.java
package com.practica.crean.arquitecture.core.usecases.impl;

import com.practica.crean.arquitecture.core.domain.Estudiante;
import com.practica.crean.arquitecture.core.port.EstudianteRepository;
import com.practica.crean.arquitecture.core.usecases.BuscarEstudiantes;

import java.util.List;

public class BuscarEstudiantesImpl implements BuscarEstudiantes {

    private EstudianteRepository repository;

    public BuscarEstudiantesImpl(EstudianteRepository repository) {
        this.repository = repository;
    }

    /*
    @Override
    public List<Estudiante> execute() {
        return repository.buscarEstudiantes();
    }
     */


}

File: src/main/kotlin/com/practica/crean/arquitecture/core/usecases/impl/BuscarEstudiantesImpl.kt
package com.example.demo.core.usecases.impl

import com.example.demo.core.domain.Estudiante
import com.example.demo.core.port.EstudianteRepository
import com.example.demo.core.usecases.BuscarEstudiantes

class BuscarEstudiantesImpl(

    private val repository: EstudianteRepository

) : BuscarEstudiantes {

    //override fun execute(): List<Estudiante> = repository.buscarEstudiantes()

}